en/developer_guide/block_parameter.rst

```diff
@@ -0,0 +1,115 @@
+Block parameter
+===============
+
+Context
+-------
+
+To be properly displayed on the website, some block will require environment parameters.
+Those parameters could be of multiple types, for instance :
+
+ - url parameters
+ - server parameters
+ - header parameters
+ - ...
+
+An easy way to give those parameters to the block would have been to give all the request
+parameters when the block is called.
+
+This easy solution is not optimal for a main reason: when loading blocks via `ESI`_, each
+block will be cached for each url variant.
+
+If the block 1 does not require any parameters, in the following exemple :
+
+ * ``/block/1`` will be in one cache entry.
+ * ``/block/1?foo=bar`` will be in another cache entry.
+
```

Comment:
It ends abruptly I think: it lacks something. It should be nice to end telling that it's not optimal to generate 2 different caches for the same html.